api/v1/controllers/lenses.js

```diff
@@ -410,18 +411,52 @@ module.exports = {
 
       updateLensDetails(seqObj);
       const assocToCreate = u.includeAssocToCreate(seqObj, helper);
-      helper.model.create(seqObj, assocToCreate)
-      .then((o) => {
-        resultObj.dbTime = new Date() - resultObj.reqStartTime;
-        delete o.dataValues.library;
-        u.logAPI(req, resultObj, o.dataValues);
-        res.status(httpStatus.CREATED).json(
-          u.responsify(o, helper, req.method)
-        );
-      })
-      .catch((err) => {
-        u.handleError(next, err, helper.modelName);
-      });
+
+      /**
+       * Creates the lens using the model.
+       * If returnUser flag is set,
+       * reloads the lens instance to return associations.
+       *
+       * @returns {Promise} The promise to create the lens.
+       */
+      const createLens = () => helper.model.create(seqObj, assocToCreate)
+        .then((o) => {
+          resultObj.dbTime = new Date() - resultObj.reqStartTime;
+          delete o.dataValues.library;
+          u.logAPI(req, resultObj, o.dataValues);
+          if (featureToggles.isFeatureEnabled('returnUser')) {
+            o.reload()
+            .then(() => res.status(httpStatus.CREATED).json(
+                u.responsify(o, helper, req.method)));
+          } else {
+            res.status(httpStatus.CREATED).json(
+              u.responsify(o, helper, req.method)
+            );
+          }
+        })
+        .catch((err) => {
+          u.handleError(next, err, helper.modelName);
+        });
+
```

Comment:
You want the lines 441 to 474 to get executed after lines 422 to 439, correct? I think the sequence is not guaranteed.